### PR TITLE
130 backend test refactoring

### LINKED ISSUE #130: Backend test refactor
**Description:**

Currently, the tests fail with minor changes to Strings or return types. We need to refactor the tests so they use the Enums instead of hardcoded string checks. We also will need to use fixtures to generate random strings rather than relying on pre-entered values as the tests should support any random values and should be robust to unseen strings

### PR DESCRIPTION
**Pull request by abhisachdev17**
### Summary of Changes
- Test refactoring was done for all of the services and blockchain
- New tests were added and failing tests were corrected
- This required some changes on the routes
- Do not worry about HttpCode or using `200` directly, I tried to be consistent with the usage but a pretty minor thing that doesn't require changes
- Some linting fixes on the backend test files

closes #130

### CODE REVIEW

File: src/backend/blockchain/src/routes.py [hunk 1]
```diff
@@ -168,13 +168,13 @@ def get_wallet_amount():
...
+    except KeyError:
         raise IncorrectPayloadException()
 
 @app.errorhandler(WalletException)
 def handle_wallet_exception(e):
     return jsonify(error=e.message), HttpCode.BAD_REQUEST.value
 
 @app.errorhandler(IncorrectPayloadException)
-def handle_wallet_exception(e):
+def handle_payload_exception(e):
```
Comment (abhisachdev17) [inline src/backend/blockchain/src/routes.py:179]: same method name, changed

File: src/backend/blockchain/tests/functional/test_routes.py [hunk 2]
```diff
@@ -26,44 +26,60 @@ def test_client():
...
+                "minerId": "user3", "proof": "already_validated", "nonce": 10}
+    return payload
+
+@pytest.fixture(scope='module')
+def test_transaction_payload():
+    return {'from': "fake_wallet_id",
+            'amount': 0,
+            'to': "another",
+            'timestamp': 0, "id": "abc", "signature": "123"}
+
```
Comment (abhisachdev17) [inline src/backend/blockchain/tests/functional/test_routes.py:52]: This is one of the major refactors, taking the stubs out of method and putting it in fixtures for reuse

File: src/backend/blockchain/tests/functional/test_routes.py [hunk 3]
```diff
@@ -114,57 +139,66 @@ def test_add_wallet(test_client):
     response = test_client.post(url, data=json.dumps(data), headers=headers)
 
     assert response.status_code == 201
-    assert response.json['success'] == True
+    assert response.json['success']
```
Comment (abhisachdev17) [inline src/backend/blockchain/tests/functional/test_routes.py:146]: lol what are we doing over here

Comment (ZackHolmberg) [reply src/backend/blockchain/tests/functional/test_routes.py:146]: That was probably from my original blockchain tests hahaha whoops

File: src/backend/blockchain/tests/functional/test_routes.py [hunk 4]
```diff
@@ -114,57 +139,66 @@ def test_add_wallet(test_client):
...
-            'amount': 0,
-            'to': "another",
-            'timestamp': 0, "id": "abc", "signature": "123"}
 
-    response = test_client.post(url, data=json.dumps(data), headers=headers)
+    response = test_client.post(url, data=json.dumps(test_transaction_payload), headers=headers)
 
     assert response.status_code == 400
-    assert b'id' in response.data
+    assert b'no corresponding wallet for id' in response.data
```
Comment (ZackHolmberg) [inline src/backend/blockchain/tests/functional/test_routes.py]: I might suggest having these return strings that we're checking for be put in variables as well. Then If they return strings change we only have to change one spot and not each individual occurrence.

**Review by ZackHolmberg — APPROVED** [review]
This is some nice work. One minor suggestion but looks great to me.

File: src/backend/blockchain/tests/functional/test_routes.py (hunk 4 above)
Comment (abhisachdev17) [reply src/backend/blockchain/tests/functional/test_routes.py]: ah, i remember noting to change it, guess i missed these. Thanks for noticing!

**Review by akshaysharma21 — APPROVED** [review]
LGTM

**Pull request merged**